logger_test.go

```diff
@@ -0,0 +1,36 @@
+package analytics
+
+import (
+	"bytes"
+	"errors"
+	"log"
+	"testing"
+)
+
+// This test ensures that the interface doesn't get changed and stays compatible
```

Comment:
Correct, I meant instead of 

```go
// This test ensures that the interface doesn't get changed and stays compatible
// with the *testing.T type.
// If someone were to modify the interface in backward incompatible manner this
// test would break.
func TestTestingLogger(t *testing.T) {
	_ = Logger(t)
}
```

we could really just do:

```go
var _ Logger = (*testing.T)(nil) // Assert interface compliance.
```